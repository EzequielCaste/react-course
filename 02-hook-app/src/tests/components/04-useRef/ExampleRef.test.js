import React from 'react'
import { shallow } from 'enzyme'
import { ExampleRef } from '../../../components/04-useRef/ExampleRef';

describe('Pruebas en <ExampleRef />', () => {

  const wrapper = shallow( <ExampleRef />);

  test('debe mostrar el componente', () => {    
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });
  
  test('debe mostrar el componente <MultipleCustomHooks />', () => {
    
    wrapper.find('button').simulate('click');
    expect( wrapper.find('MultipleCustomHooks').exists()).toBe(true);

  })
})