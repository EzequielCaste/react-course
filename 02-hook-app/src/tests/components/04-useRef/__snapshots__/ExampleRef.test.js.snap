// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ExampleRef /> debe mostrar el componente 1`] = `
<div>
  <h1>
    Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
